apps/agreement/contracts/Agreement.sol

```diff
@@ -424,148 +365,154 @@ contract Agreement is IArbitrable, IForwarder, AragonApp {
         (Action storage action, Dispute storage dispute) = _getActionAndDispute(_disputeId);
         require(_canRuleDispute(action), ERROR_CANNOT_RULE_ACTION);
 
-        Setting storage setting = _getSetting(action);
         address arbitratorAddress = address(arbitrator);
         require(msg.sender == arbitratorAddress, ERROR_SENDER_NOT_ALLOWED);
 
         dispute.ruling = _ruling;
         emit Ruled(IArbitrator(arbitratorAddress), _disputeId, _ruling);
 
         if (_ruling == DISPUTES_RULING_SUBMITTER) {
-            _rejectChallenge(action, setting);
+            _rejectChallenge(action);
             emit ActionAccepted(dispute.actionId);
```

Comment:
Why are we leaving the events out of these 3 internal functions? Is it to avoid passing the id?

Reply:
Yes